patches/kaggle_secrets.py

```diff
@@ -0,0 +1,59 @@
+import json
+import os
+import urllib.request
+
+_KAGGLE_DEFAULT_URL_BASE = "https://www.kaggle.com"
+_KAGGLE_URL_BASE_ENV_VAR_NAME = "KAGGLE_URL_BASE"
+_KAGGLE_USER_SECRETS_TOKEN_ENV_VAR_NAME = "KAGGLE_USER_SECRETS_TOKEN_KEY"
+
+
+class CredentialError(Exception):
+    pass
+
+
+class BackendError(Exception):
+    pass
+
+
+class UserSecretsClient():
+    GET_USER_SECRET_ENDPOINT = '/requests/GetUserSecretRequest'
+    BIGQUERY_PURPOSE_VALUE = 1
+
+    def __init__(self):
+        url_base_override = os.getenv(_KAGGLE_URL_BASE_ENV_VAR_NAME)
+        self.url_base = (url_base_override or _KAGGLE_DEFAULT_URL_BASE)
+        # Follow the OAuth 2.0 Authorization standard (https://tools.ietf.org/html/rfc6750)
+        jwt_token = os.getenv(_KAGGLE_USER_SECRETS_TOKEN_ENV_VAR_NAME)
+        if jwt_token is None:
+            raise CredentialError(
+                f'A JWT Token is required to use the UserSecretsClient, but none found in environment variable {_KAGGLE_USER_SECRETS_TOKEN_ENV_VAR_NAME}')
+        self.headers = {'Content-type': 'application/json',
+                        'Authorization': 'Bearer {}'.format(jwt_token)}
+
+    def _make_get_request(self, request_body):
+        request_params = urllib.parse.urlencode(request_body)
+        url = f'{self.url_base}{self.GET_USER_SECRET_ENDPOINT}?{request_params}'
+        req = urllib.request.Request(url, headers=self.headers)
+        with urllib.request.urlopen(req) as response:
+            response_json = json.loads(response.read())
+            return response_json
+
+    def getUserSecret(self, secret_label: str):
+        request_body = {
+            'SecretLabel': secret_label
```

Comment:
What is 'label' here? I think we were talking about something like "purpose" instead, and I see you use it below.

Reply:
Yep, purpose is used for specific integrations like bigquery/gcs, where there's no specific "label" the user has given their secret (since it's a refresh token managed by us). In the case of the general usre secret, we'll require a label or name that the user gives the user secret. I know we don't support it yet, but I figured it was easy enough to add and show's the use case. Happy to rename it if we come up with a different name for it.

Reply:
I'd remove this one for now. Label indicate something user visible (and also entered by the user). If we are going to be doing a lookup like that in the future, I'd still think we'll use some sort of ID (or name, like you said), that may be associated with a label, but not a label itself.

Reply:
We just discussed over slack, but adding here, just to close this comment: because your change is user visible (in a sense), my recommendation would be to add this later once we figure out how exactly we support general case (with proper validation, etc.). I agree that we should try to make sure the general use case is easy to enable.